app/src/main/java/com/example/sharingang/ItemsListFragment.kt

```diff
@@ -42,8 +43,20 @@ class ItemsListFragment : Fragment() {
             goToMap(view)
         }
         binding.userProfileButton.setOnClickListener {
+            val sharedPreferences = activity?.getSharedPreferences(
+                getString(R.string.preference_user_info),
+                Context.MODE_PRIVATE
+            )
             val action =
+                if (sharedPreferences == null) {
                     ItemsListFragmentDirections.actionItemsListFragmentToUserProfileFragment("test")
+                } else {
+                    ItemsListFragmentDirections.actionItemsListFragmentToUserProfileFragment(
+                        sharedPreferences.getString(
+                            getString(R.string.account_firebase_uid), "test"
```

Comment:
Should we keep "test" as a default string? I don't know what else to put but maybe test isn't the best.